clouddriver-aws/src/main/java/com/netflix/spinnaker/clouddriver/aws/deploy/ops/CopyLastAsgAtomicOperation.java

```diff
@@ -0,0 +1,552 @@
+                    Boolean instanceMonitoring = null;
+                    Boolean associatePublicIpAddress = null;
+
+                    List<String> securityGroups;
+                    List<String> classicLinkVPCSecurityGroups = null;
+                    if (ancestorAsg.getLaunchTemplate() != null) {
+                      LaunchTemplateSpecification launchTemplate = ancestorAsg.getLaunchTemplate();
+                      LaunchTemplateVersion launchTemplateVersion =
+                          sourceRegionScopedProvider
+                              .getLaunchTemplateService()
+                              .getLaunchTemplateVersion(ancestorAsg.getLaunchTemplate())
+                              .orElseThrow(
+                                  () ->
+                                      new IllegalStateException(
+                                          "Requested launch template "
+                                              + launchTemplate
+                                              + " was not found"));
+
+                      ResponseLaunchTemplateData launchTemplateData =
+                          launchTemplateVersion.getLaunchTemplateData();
+                      newDescription.setSetLaunchTemplate(true);
+                      imageId = launchTemplateData.getImageId();
+                      keyName = launchTemplateData.getKeyName();
+                      kernelId = launchTemplateData.getKernelId();
+                      userData = launchTemplateData.getUserData();
+                      ramdiskId = launchTemplateData.getRamDiskId();
+                      instanceType = launchTemplateData.getInstanceType();
+                      securityGroups = launchTemplateData.getSecurityGroups();
+                      ebsOptimized = launchTemplateData.getEbsOptimized();
+                      if (launchTemplateData.getIamInstanceProfile() != null) {
+                        iamInstanceProfile = launchTemplateData.getIamInstanceProfile().getName();
+                      }
+
+                      if (launchTemplateData.getMonitoring() != null) {
+                        instanceMonitoring = launchTemplateData.getMonitoring().getEnabled();
+                      }
+
+                      if (launchTemplateData.getInstanceMarketOptions() != null) {
+                        LaunchTemplateSpotMarketOptions options =
+                            launchTemplateData.getInstanceMarketOptions().getSpotOptions();
+                        if (options != null) {
+                          spotPrice = options.getMaxPrice();
+                        }
+                      }
+
+                      if (launchTemplateData.getMetadataOptions() != null) {
+                        newDescription.setRequireIMDSv2(
+                            launchTemplateData
+                                .getMetadataOptions()
+                                .getHttpTokens()
+                                .equals("required"));
```

Comment:
Is it possible that `getHttpTokens()` can return null? Maybe more better to `"required".equals(...)` instead.